File: src/client/java/teammates/client/scripts/scalabilitytests/instructorCourseEnrollPageScaleTestDataGenerator.py
import names

FILENAME = 'InstructorCourseEnrollPageScaleTestData'

HEADER = "Section | Team | Name | Email | Comments\n"


def generate_students(num):
    student_names = []
    students = HEADER
    for i in range(num):
        cur_name = names.get_full_name()
        while cur_name in student_names:
            cur_name = names.get_full_name()
        cur_username = cur_name.lower().replace(" ", ".")
        students += ('Section ' + str(i // 100 + 1) + ' | Team '
                     + str(i // 100 + 1) + ' | ' + cur_name + ' | '
                     + cur_username + '.tmms@gmail.tmt' + ' | \n')
    return students


student_groups = [10, 20, 50, 75, 100, 150]

for i in student_groups:
    with open(FILENAME + str(i), 'w') as file:
        file.write(generate_students(i))

File: src/client/java/teammates/client/scripts/scalabilitytests/InstructorCourseEnrollPageScaleTestDataGenerator.java
package teammates.client.scripts.scalabilitytests;

import org.kohsuke.randname.RandomNameGenerator;

import java.io.File;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.HashSet;

/**
 * Generates test data for InstructorCourseEnrollPageScaleTest.
 */
public final class InstructorCourseEnrollPageScaleTestDataGenerator {

    private static final String FILENAME = "InstructorCourseEnrollPageScaleTestData";

    private static final String HEADER = "Section | Team | Name | Email | Comments\n";

    private static final int[] studentGroups = {10, 20, 50, 75, 100, 150};

    private InstructorCourseEnrollPageScaleTestDataGenerator() {
    }

    private static String generateStudents(int num) {
        RandomNameGenerator generator = new RandomNameGenerator(num);
        HashSet<String> studentNames = new HashSet<>();
        StringBuffer students = new StringBuffer(HEADER);
        for (int i = 0; i < num; i++) {
            String curName = generator.next();
            while (studentNames.contains(curName)) {
                curName = generator.next();
            }

            students.append(
                    String.format("Section %d | Team %d | %s | %s.tmms@gmail.tmt | %n",
                            i / 100 + 1, i / 100 + 1, curName.replace("_", " "), curName.replace("_", ".")));
        }
        return students.toString();
    }

    public static void main(String[] args) throws IOException {
        for (int studentGroup : studentGroups) {
            File file = new File("src/client/java/teammates/client/scripts/scalabilitytests/" + FILENAME + studentGroup);
            try (PrintWriter out = new PrintWriter(file)) {
                out.println(generateStudents(studentGroup));
                out.close();
            }
        }
    }
}
